tests/driver/api.test.ts

```diff
@@ -0,0 +1,633 @@
+            } catch(err: OperationNotSupportedError | any) {
+                //console.log('Exception caught: ', err);
+                //TODO - need to figure out how to test this
+                error = err;
+            } finally {
+                if(autoIndexStatus != undefined) {
+                    mongooseInstance!.set('autoIndex', autoIndexStatus);
+                }
+                await dropCollections(isAstra, astraMongoose, jsonAPIMongoose, 'users');
+            }
+            assert.ok(error);
+            assert.strictEqual(error?.message, 'createIndex() Not Implemented');
+        });
+        it('API ops tests Model.insertMany()', async () => {
+            const product1 = {name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'};
+            const product2 = {name: 'Product 2', price: 10, isCertified: true, category: 'cat 2'};
+            const product3 = {name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'};
+            const insertResp = await Product.insertMany([product1, product2, product3] , {ordered: true});
+            assert.strictEqual(insertResp.length, 3);
+            assert.strictEqual(insertResp[0].name, 'Product 1');
+            assert.strictEqual(insertResp[1].name, 'Product 2');
+            assert.strictEqual(insertResp[2].name, 'Product 3');
+        });
+        //Model.inspect can not be tested since its a helper for console logging. More info here: https://mongoosejs.com/docs/api/model.html#Model.inspect()
+        it('API ops tests Model.listIndexes()', async () => {
+            let error: OperationNotSupportedError | null = null;
+            try {
+                const listIndexesResp = await Product.listIndexes();
+            } catch(err: OperationNotSupportedError | any) {
+                error = err;
+            }
+            assert.ok(error);
+            assert.strictEqual(error?.message, 'listIndexes() Not Implemented');
+        });
+        it('API ops tests Model.populate()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'});
+            const product2 = new Product({name: 'Product 2', price: 10, isCertified: true, category: 'cat 2'});
+            const product3 = new Product({name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'});
+            await Product.insertMany([product1, product2, product3]);
+            const cart1 = new Cart({name: 'Cart 1', products: [product1._id, product2._id]});
+            await Cart.insertMany([cart1]);
+            const populateResp = await Cart.findOne({name: 'Cart 1'}).populate('products');
+            assert.strictEqual(populateResp?.products.length, 2);
+            assert.strictEqual(populateResp?.products[0].name, 'Product 1');
+            assert.strictEqual(populateResp?.products[1].name, 'Product 2');
+        });
+        //TODO - should we test all Model.prototype. methods as well?
```

Comment:
I'd recommend testing `Model.prototype.deleteOne()`, the rest of `Model.prototype` methods would not be useful to test.

Reply:
Sure, added test only for `deleteOne()` in `Model.prototype` APIs